doc/source/whatsnew/v1.0.0.rst

```diff
@@ -451,6 +451,7 @@ Reshaping
 - Fix to ensure all int dtypes can be used in :func:`merge_asof` when using a tolerance value. Previously every non-int64 type would raise an erroneous ``MergeError`` (:issue:`28870`).
 - Better error message in :func:`get_dummies` when `columns` isn't a list-like value (:issue:`28383`)
 - Bug :meth:`Series.pct_change` where supplying an anchored frequency would throw a ValueError (:issue:`28664`)
+- Bug where :meth:`DataFrame.equals` returned True incorrectly in some circumstances(:issue:`28839`)
```

Comment:
The `some circumstances` is really just that "order was ignored for same-dtyped columns" right? _Might_ be good to clarify further if possible

Reply:
Will message 
"- Bug where :meth:`DataFrame.equals` returned True incorrectly in some cases when two DataFrames had the same columns but in different orders" 
be better?

Reply:
Yea I think that's great!

Reply:
Done.